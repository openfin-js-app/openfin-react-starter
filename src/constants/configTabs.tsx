import * as React from 'react';
import InputAdornment from '@material-ui/core/InputAdornment';

import { IConfigTab, MuiTheme, FieldType } from 'react-openfin';
import {
    ConfigAboutField,
    ConfigThemeField,
    ConfigLangField,
    ConfigJsonField,
    ConfigView,
} from 'react-openfin-mat-impl';

import SettingIcon from '@material-ui/icons/Settings';

import i18n from '../i18n';

import appSvg from '../assets/svg/app.svg';
import companySvg from '../assets/svg/company.svg';
import controlsSvg from'../assets/svg/support/controls.svg';
import controlsDarkSvg from'../assets/svg/support/controls_dark.svg';
import infomationSvg from '../assets/svg/support/information.svg';
import infomationDarkSvg from '../assets/svg/support/information-dark.svg';
import listCheckedSvg from'../assets/svg/other/check-box.svg';
import listCheckedDarkSvg from'../assets/svg/other/list-checked-dark.svg';

function _getSampleDate() {
    return process.env.NODE_ENV === 'test'? new Date(2018,9,13,16,53,41): new Date();
}

const configTabs:IConfigTab[]=[
    {
        _order: 0 ,
        _label: 'application.label',
        _name:'application',
        _svgUrl: controlsSvg,
        _svgUrl_dark: controlsDarkSvg,
        _icon: SettingIcon,
        _fields:[
            {
                _type:FieldType.SUBHEADING,
                _label:'application.theme.label',
                _cols: 2,
                _rows:1,
            },
            {
                _type:FieldType.CUSTOM_FIELD,
                _label:'Theme',
                _custom:<ConfigThemeField/>,
                _name:'theme',
                _defaultValue:MuiTheme.DARK,
                _cols:10,
                _rows:1,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'application.language.label',
                _cols: 2,
                _rows:1,
            },
            {
                _type:FieldType.CUSTOM_FIELD,
                _label:'language',
                _custom:<ConfigLangField/>,
                _name:'language',
                _defaultValue:i18n.language,
                _cols:10,
                _rows:1,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'application.newWinPos.label',
                _cols: 12,
            },
            {
                _type:FieldType.NUMBER,
                _label:'application.newWinPos.newWinTop',
                _name:'newWinTop',
                _defaultValue:parseInt(process.env.REACT_APP_NEW_WINDOW_TOP,10),
                _props:{
                    InputProps:{
                        endAdornment:<InputAdornment position={'end'}>Pixel</InputAdornment>
                    }
                }
            },
            {
                _type:FieldType.NUMBER,
                _label:'application.newWinPos.newWinLeft',
                _name:'newWinLeft',
                _defaultValue:parseInt(process.env.REACT_APP_NEW_WINDOW_LEFT,10),
                _props:{
                    InputProps:{
                        endAdornment:<InputAdornment position={'end'}>Pixel</InputAdornment>
                    }
                }
            },
            {
                _type:FieldType.NUMBER,
                _label:'application.newWinPos.newWinWidth',
                _name:'newWinWidth',
                _defaultValue:parseInt(process.env.REACT_APP_NEW_WINDOW_WIDTH,10),
                _props:{
                    InputProps:{
                        endAdornment:<InputAdornment position={'end'}>Pixel</InputAdornment>
                    }
                }
            },
            {
                _type:FieldType.NUMBER,
                _label:'application.newWinPos.newWinHeight',
                _name:'newWinHeight',
                _defaultValue:parseInt(process.env.REACT_APP_NEW_WINDOW_HEIGHT,10),
                _props:{
                    InputProps:{
                        endAdornment:<InputAdornment position={'end'}>Pixel</InputAdornment>
                    }
                }
            },
            {
                _type:FieldType.NUMBER,
                _label:'application.newWinPos.newWindDeltaLeft',
                _name:'newWindDeltaLeft',
                _defaultValue:parseInt(process.env.REACT_APP_NEW_WINDOW_DELTA_LEFT,10),
                _props:{
                    InputProps:{
                        endAdornment:<InputAdornment position={'end'}>Pixel</InputAdornment>
                    }
                }
            },
            {
                _type:FieldType.NUMBER,
                _label:'application.newWinPos.newWindDeltaHeight',
                _name:'newWindDeltaHeight',
                _defaultValue:parseInt(process.env.REACT_APP_NEW_WINDOW_DELTA_HEIGHT,10),
                _props:{
                    InputProps:{
                        endAdornment:<InputAdornment position={'end'}>Pixel</InputAdornment>
                    }
                }
            },
        ]
    },
    {
        _order: 1 ,
        _label: 'about.label',
        _name:'about',
        _svgUrl: infomationSvg,
        _svgUrl_dark: infomationDarkSvg,
        _icon: SettingIcon,
        _fields:[
            {
                _type:FieldType.CUSTOM_FIELD,
                _label:'About openfin starter',
                _custom:
                    <ConfigAboutField
                        appLogo={appSvg}
                        companyLogo={companySvg}
                    />
                ,
                _cols:12,
                _rows:12,
            }
        ]
    },
    {
        _order: 2,
        _label: 'sample.label',
        _name:'sample',
        _svgUrl: listCheckedSvg,
        _svgUrl_dark: listCheckedDarkSvg,
        _icon: SettingIcon,
        _fields:[
            {
                _type:FieldType.SUBHEADING,
                _label:'sample.vals.label',
                _cols: 12,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'sample.vals.string.label',
                _cols: 4,
            },
            {
                _type:FieldType.STRING,
                _label:'sample.vals.string.field',
                _name:'strValue',
                _defaultValue:"",
                _cols: 8,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'sample.vals.currency.label',
                _cols: 4,
            },
            {
                _type:FieldType.CURRENCY,
                _label:'sample.vals.currency.field',
                _name:'crrValue',
                _defaultValue:123456789,
                _cols: 8,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'sample.vals.date.label',
                _cols: 4,
            },
            {
                _type:FieldType.DATE,
                _label:'sample.vals.date.field',
                _name:'dateValue',
                _defaultValue:_getSampleDate(),
                _cols: 8,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'sample.vals.time.label',
                _cols: 4,
            },
            {
                _type:FieldType.TIME,
                _label:'sample.vals.time.field',
                _name:'timeValue',
                _defaultValue:_getSampleDate(),
                _cols: 8,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'sample.vals.dateTime.label',
                _cols: 4,
            },
            {
                _type:FieldType.DATETIME,
                _label:'sample.vals.dateTime.field',
                _name:'datetimeValue',
                _defaultValue:_getSampleDate(),
                _cols: 8,
            },
            {
                _type:FieldType.SUBHEADING,
                _label:'sample.vals.customField.label',
                _cols: 12,
            },
            {
                _type:FieldType.CUSTOM_FIELD,
                _label:'sample.vals.customField.field',
                _custom:<ConfigJsonField/>,
                _cols:12,
                _rows:6,
            }
        ]
    }
];

export const MyConfigView:React.FunctionComponent<{}> = ({})=>{
    return (<ConfigView tabs={configTabs}/>)
}

export default configTabs;
