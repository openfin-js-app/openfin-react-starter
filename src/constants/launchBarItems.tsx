import {ILaunchBarItem} from 'react-openfin';

import AccessibilityIcon from '@material-ui/icons/Accessibility';
import AccountBoxIcon from '@material-ui/icons/AccountBox';
import BookmarkIcon from '@material-ui/icons/Bookmark';
import BackupIcon from '@material-ui/icons/Backup';
import BugReportIcon from '@material-ui/icons/BugReport';
import LanguageIcon from '@material-ui/icons/Language';
import ReportIcon from '@material-ui/icons/Report';

import svgNumberOne from '../assets/svg/other/number-1.svg';
import svgNumberTwo from '../assets/svg/other/number-2.svg';
import googleSearch from '../assets/svg/other/google-search.svg';

export const sampleItems:ILaunchBarItem[]=[
    {
        icon:AccessibilityIcon,
        disabled:false,
        svg:null,
        appJson:{
            name:`${process.env.REACT_APP_FIN_UUID}-child-accessibility`,
            url:`${process.env.PUBLIC_URL}/childWindow/accessibility`,
            frame:false,
            resizable:true,
            state:'normal',
            autoShow:true,
            minWidth:520,
            minHeight:170,
        },
    },
    {
        icon:null,
        disabled:false,
        svg:svgNumberOne,
        appJson:{
            name:`${process.env.REACT_APP_FIN_UUID}-child-view-one`,
            url:`${process.env.PUBLIC_URL}/childWindow/view-one`,
            frame:false,
            resizable:true,
            state:'normal',
            autoShow:true,
            minWidth:415,
            minHeight:100,
        },
    },
    {
        icon:null,
        disabled:false,
        svg:svgNumberTwo,
        appJson:{
            name:`${process.env.REACT_APP_FIN_UUID}-child-view-two`,
            url:`${process.env.PUBLIC_URL}/childWindow/view-two`,
            frame:false,
            resizable:true,
            state:'normal',
            autoShow:true,
            minWidth:415,
            minHeight:100,
        },
        // !!!README!!!
        // to hide a launch bar item like basing on entitlement,
        // set shown field to false or a cb returning false
        // shown:()=>{return false},
    },
    {
        icon:ReportIcon,
        disabled:false,
        svg:null,
        appJson:{
            name:`${process.env.REACT_APP_FIN_UUID}-child-report`,
            url:`${process.env.PUBLIC_URL}/childWindow/report`,
            frame:false,
            resizable:true,
            state:'normal',
            autoShow:true,
            minWidth:430,
            minHeight:100,
        },
    },
    {
        icon:null,
        disabled:false,
        svg:googleSearch,
        appJson:{
            name:`${process.env.REACT_APP_FIN_UUID}-child-google`,
            url:'https://www.google.com/',
            frame:true,
            resizable:true,
            state:'normal',
            autoShow:true,
        },
    },
];

export const disabledItems:ILaunchBarItem[]=[
    {
        icon:AccountBoxIcon,
        disabled:true,
        svg:null,
        appJson:{},
    },
    {
        icon:BookmarkIcon,
        disabled:true,
        svg:null,
        appJson:{},
    },
    {
        icon:BackupIcon,
        disabled:true,
        svg:null,
        appJson:{},
    },
    {
        icon:BugReportIcon,
        disabled:true,
        svg:null,
        appJson:{},
    },
    {
        icon:LanguageIcon,
        disabled:true,
        svg:null,
        appJson:{},
    },
];

export const launchBarItems = [
    ...sampleItems,
    ...disabledItems,
];

export default launchBarItems;