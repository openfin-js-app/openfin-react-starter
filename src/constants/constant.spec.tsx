import * as React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import configTabs, { MyConfigView } from './configTabs';
import launchBarItems from './launchBarItems';

describe('global constants',()=>{

    let shallow;
    let mount;

    beforeEach(() => {
        shallow = createShallow();
        mount = createMount();
    });

    afterEach(()=>{
        mount.cleanUp();
    })

    it('config tabs',()=>{
        expect(configTabs).toMatchSnapshot();
    })

    it('MyConfigView',()=>{
        const wrapper = shallow(<MyConfigView/>)
        expect(wrapper).toBeTruthy();
    })

    it('launchbar items',()=>{
        expect(launchBarItems).toMatchSnapshot();
    })

})