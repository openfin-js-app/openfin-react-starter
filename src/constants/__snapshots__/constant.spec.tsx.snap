// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global constants config tabs 1`] = `
Array [
  Object {
    "_fields": Array [
      Object {
        "_cols": 2,
        "_label": "application.theme.label",
        "_rows": 1,
        "_type": 9,
      },
      Object {
        "_cols": 10,
        "_custom": <ConfigThemeField />,
        "_defaultValue": "dark",
        "_label": "Theme",
        "_name": "theme",
        "_rows": 1,
        "_type": 1,
      },
      Object {
        "_cols": 2,
        "_label": "application.language.label",
        "_rows": 1,
        "_type": 9,
      },
      Object {
        "_cols": 10,
        "_custom": <ConfigLangField />,
        "_defaultValue": undefined,
        "_label": "language",
        "_name": "language",
        "_rows": 1,
        "_type": 1,
      },
      Object {
        "_cols": 12,
        "_label": "application.newWinPos.label",
        "_type": 9,
      },
      Object {
        "_defaultValue": 60,
        "_label": "application.newWinPos.newWinTop",
        "_name": "newWinTop",
        "_props": Object {
          "InputProps": Object {
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              Pixel
            </ForwardRef(WithStyles)>,
          },
        },
        "_type": 6,
      },
      Object {
        "_defaultValue": 300,
        "_label": "application.newWinPos.newWinLeft",
        "_name": "newWinLeft",
        "_props": Object {
          "InputProps": Object {
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              Pixel
            </ForwardRef(WithStyles)>,
          },
        },
        "_type": 6,
      },
      Object {
        "_defaultValue": 640,
        "_label": "application.newWinPos.newWinWidth",
        "_name": "newWinWidth",
        "_props": Object {
          "InputProps": Object {
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              Pixel
            </ForwardRef(WithStyles)>,
          },
        },
        "_type": 6,
      },
      Object {
        "_defaultValue": 320,
        "_label": "application.newWinPos.newWinHeight",
        "_name": "newWinHeight",
        "_props": Object {
          "InputProps": Object {
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              Pixel
            </ForwardRef(WithStyles)>,
          },
        },
        "_type": 6,
      },
      Object {
        "_defaultValue": 20,
        "_label": "application.newWinPos.newWindDeltaLeft",
        "_name": "newWindDeltaLeft",
        "_props": Object {
          "InputProps": Object {
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              Pixel
            </ForwardRef(WithStyles)>,
          },
        },
        "_type": 6,
      },
      Object {
        "_defaultValue": 20,
        "_label": "application.newWinPos.newWindDeltaHeight",
        "_name": "newWindDeltaHeight",
        "_props": Object {
          "InputProps": Object {
            "endAdornment": <ForwardRef(WithStyles)
              position="end"
            >
              Pixel
            </ForwardRef(WithStyles)>,
          },
        },
        "_type": 6,
      },
    ],
    "_icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "SettingsIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "_label": "application.label",
    "_name": "application",
    "_order": 0,
    "_svgUrl": "controls.svg",
    "_svgUrl_dark": "controls_dark.svg",
  },
  Object {
    "_fields": Array [
      Object {
        "_cols": 12,
        "_custom": <ConfigAboutField
          appLogo="app.svg"
          companyLogo="company.svg"
        />,
        "_label": "About openfin starter",
        "_rows": 12,
        "_type": 1,
      },
    ],
    "_icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "SettingsIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "_label": "about.label",
    "_name": "about",
    "_order": 1,
    "_svgUrl": "information.svg",
    "_svgUrl_dark": "information-dark.svg",
  },
  Object {
    "_fields": Array [
      Object {
        "_cols": 12,
        "_label": "sample.vals.label",
        "_type": 9,
      },
      Object {
        "_cols": 4,
        "_label": "sample.vals.string.label",
        "_type": 9,
      },
      Object {
        "_cols": 8,
        "_defaultValue": "",
        "_label": "sample.vals.string.field",
        "_name": "strValue",
        "_type": 2,
      },
      Object {
        "_cols": 4,
        "_label": "sample.vals.currency.label",
        "_type": 9,
      },
      Object {
        "_cols": 8,
        "_defaultValue": 123456789,
        "_label": "sample.vals.currency.field",
        "_name": "crrValue",
        "_type": 7,
      },
      Object {
        "_cols": 4,
        "_label": "sample.vals.date.label",
        "_type": 9,
      },
      Object {
        "_cols": 8,
        "_defaultValue": 2018-10-13T20:53:41.000Z,
        "_label": "sample.vals.date.field",
        "_name": "dateValue",
        "_type": 4,
      },
      Object {
        "_cols": 4,
        "_label": "sample.vals.time.label",
        "_type": 9,
      },
      Object {
        "_cols": 8,
        "_defaultValue": 2018-10-13T20:53:41.000Z,
        "_label": "sample.vals.time.field",
        "_name": "timeValue",
        "_type": 5,
      },
      Object {
        "_cols": 4,
        "_label": "sample.vals.dateTime.label",
        "_type": 9,
      },
      Object {
        "_cols": 8,
        "_defaultValue": 2018-10-13T20:53:41.000Z,
        "_label": "sample.vals.dateTime.field",
        "_name": "datetimeValue",
        "_type": 3,
      },
      Object {
        "_cols": 12,
        "_label": "sample.vals.customField.label",
        "_type": 9,
      },
      Object {
        "_cols": 12,
        "_custom": <ConfigJsonField />,
        "_label": "sample.vals.customField.field",
        "_rows": 6,
        "_type": 1,
      },
    ],
    "_icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "SettingsIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "_label": "sample.label",
    "_name": "sample",
    "_order": 2,
    "_svgUrl": "check-box.svg",
    "_svgUrl_dark": "list-checked-dark.svg",
  },
]
`;

exports[`global constants launchbar items 1`] = `
Array [
  Object {
    "appJson": Object {
      "autoShow": true,
      "frame": false,
      "minHeight": 170,
      "minWidth": 520,
      "name": "openfin_react_ts_starter_test-child-accessibility",
      "resizable": true,
      "state": "normal",
      "url": "/childWindow/accessibility",
    },
    "disabled": false,
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "AccessibilityIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "svg": null,
  },
  Object {
    "appJson": Object {
      "autoShow": true,
      "frame": false,
      "minHeight": 100,
      "minWidth": 415,
      "name": "openfin_react_ts_starter_test-child-view-one",
      "resizable": true,
      "state": "normal",
      "url": "/childWindow/view-one",
    },
    "disabled": false,
    "icon": null,
    "svg": "number-1.svg",
  },
  Object {
    "appJson": Object {
      "autoShow": true,
      "frame": false,
      "minHeight": 100,
      "minWidth": 415,
      "name": "openfin_react_ts_starter_test-child-view-two",
      "resizable": true,
      "state": "normal",
      "url": "/childWindow/view-two",
    },
    "disabled": false,
    "icon": null,
    "svg": "number-2.svg",
  },
  Object {
    "appJson": Object {
      "autoShow": true,
      "frame": false,
      "minHeight": 100,
      "minWidth": 430,
      "name": "openfin_react_ts_starter_test-child-report",
      "resizable": true,
      "state": "normal",
      "url": "/childWindow/report",
    },
    "disabled": false,
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "ReportIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "svg": null,
  },
  Object {
    "appJson": Object {
      "autoShow": true,
      "frame": true,
      "name": "openfin_react_ts_starter_test-child-google",
      "resizable": true,
      "state": "normal",
      "url": "https://www.google.com/",
    },
    "disabled": false,
    "icon": null,
    "svg": "google-search.svg",
  },
  Object {
    "appJson": Object {},
    "disabled": true,
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "AccountBoxIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "svg": null,
  },
  Object {
    "appJson": Object {},
    "disabled": true,
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "BookmarkIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "svg": null,
  },
  Object {
    "appJson": Object {},
    "disabled": true,
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "BackupIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "svg": null,
  },
  Object {
    "appJson": Object {},
    "disabled": true,
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "BugReportIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "svg": null,
  },
  Object {
    "appJson": Object {},
    "disabled": true,
    "icon": Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "displayName": "LanguageIcon",
      "muiName": "SvgIcon",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "svg": null,
  },
]
`;
